### PR TITLE
Asher dl

### PR DESCRIPTION
**Pull request by AsherDLL**
**Pull request fixed**

Please add the correspondig button to land in my view from the main page(for root users only)

### CODE REVIEW

File: app/Http/Controllers/PermissionsController.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+use App\UserType;
+use App\User;
+
+class PermissionsController extends Controller 
+{
+
+
+	public function getUser()
+	{
+		if(!Auth::check())
```
Comment (rikyfocil-old) [inline app/Http/Controllers/PermissionsController.php:17]: Next time place something like this

```
function __construct(){
    $this->middleware('auth');
}
```

With it only authenticated users can acces the route and non authenticated users will go to login.

File: app/Http/Controllers/PermissionsController.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+			return redirect()->route('home');
+
+		if(Auth::user()->idUserType != 3)
+			return redirect()->route('home');
```
Comment (rikyfocil-old) [inline app/Http/Controllers/PermissionsController.php:21]: Can you change this for 
`abort(403);`
?

File: app/Http/Controllers/PermissionsController.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	}
+
+	public function changePermission(Request $request, $id)
+	{
+		if(!Auth::check())
+			return redirect()->route('home');
+
+		if(Auth::user()->idUserType != 3)
+			return redirect()->route('home');
```
Comment (rikyfocil-old) [inline app/Http/Controllers/PermissionsController.php:33]: Same as above

File: app/Http/Controllers/PermissionsController.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+		$change_permission = User::findOrFail($id);
+		$change_permission->idUserType = $request["permission"];
```
Comment (rikyfocil-old) [inline app/Http/Controllers/PermissionsController.php:36]: Please validate the request
https://laravel.com/docs/5.3/validation

File: app/User.php [hunk 5]
```diff
@@ -35,11 +35,19 @@ public function sharedFiles(){
         return $this->hasMany('App\Share', 'idUser');
     }
 
+
+     public function permission(){
+        return $this->hasOne('App\UserType', 'id', 'idUserType');
```
Comment (rikyfocil-old) [inline app/User.php:40]: Still need to be changed for belongs

**Review by rikyfocil-old — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/PermissionsController.php (hunk 1 above)
Comment (AsherDLL) [reply app/Http/Controllers/PermissionsController.php:17]: Ok

File: app/User.php (hunk 5 above)
Comment (AsherDLL) [reply app/User.php:40]: Fixed

File: app/Http/Controllers/PermissionsController.php (hunk 2 above)
Comment (AsherDLL) [reply app/Http/Controllers/PermissionsController.php:21]: Fixed

File: app/Http/Controllers/PermissionsController.php (hunk 3 above)
Comment (AsherDLL) [reply app/Http/Controllers/PermissionsController.php:33]: Fixed

File: app/Http/Controllers/PermissionsController.php (hunk 4 above)
Comment (AsherDLL) [reply app/Http/Controllers/PermissionsController.php:36]: Fixed

**Pull request closed without merging**